MetaMask/metamask-extension
# [FLASK] Expose transaction origin to transaction insight snaps

## Explanation

Allows transaction insight snaps to access the origin of a transaction by specifying a permission caveat.

This PR also reworks the permission description system to support each permission resulting in multiple "permission descriptions"

Fixes #16631 

## Manual Testing Steps
1. Create a transaction insight snap with the permission `endowment:transaction-insight: { allowTransactionOrigin: true }`
2. Install the snap
3. Propose a transaction
4. See that the snap can access the transaction origin
